homeassistant/components/geo_location/__init__.py

```diff
@@ -64,32 +71,38 @@ def state(self):
     @property
     def source(self) -> str:
         """Return source value of this external event."""
+        if hasattr(self, "_attr_source"):
+            return self._attr_source
         raise NotImplementedError
```

Comment:
This is enough, as it will raise an attribute error instead. We've implemented these occurrences in the same fashion as with other platforms.

```suggestion
        return self._attr_source
```